frontend/src/pages/AnnouncementPage.js

```diff
@@ -0,0 +1,420 @@
+	const [errorMessageGetAnnouncement, setErrorMessageGetAnnouncement] = useState("");
+	const [userIsAdmin, setUserIsAdmin] = useState(false);
+
+	const [editing, setEditing] = useState("");
+
+	const getAnnouncementsFromServer = () => {
+		if (!announcementsAreUpToDate) {
+			appwriteApi.getAnnouncements()
+				.then((result) => {
+					setAnnouncementsAreUpToDate(true);
+					setAnnouncementsFromServer(result.documents);
+				})
+				.catch((e) => {
+					setErrorMessageGetAnnouncement("Error getting announcement from server:");
+					console.log(e);
+				});
+		}
+	};
+
+	useEffect(() => {
+		getAnnouncementsFromServer();
+		if (user && user.user) {
+			appwriteApi.userIsMemberOfTeam("Admins")
+				.then(isAdmin => setUserIsAdmin(isAdmin));
+		} else {
+			setUserIsAdmin(false);
+		}
+	});
+
+	const clearInputFields = () => {
+		document.getElementById("titleInputText").value = "";
+		document.getElementById("contentInputText").value = "";
+	};
+
+	const handleClearButton = () => {
+		clearInputFields();
+	};
+
+	const handleSubmitButton = () => {
+		const title = document.getElementById("titleInputText");
+		const content = document.getElementById("contentInputText");
+		if (title.value.length == 0 || content.value.length == 0) {
+			console.log("missing input or content!");
+			return;
+		}
+		appwriteApi.createAnnouncement({
+			"title": title.value,
+			"content": content.value,
+			"created_at": new Date().valueOf()
+		})
+			.then(() => {
+				setAnnouncementsAreUpToDate(false);
+				clearInputFields();
+			})
+			.catch((e) => {
+				setErrorMessageAddAnnouncement("Error adding announcement to server");
+				console.log(e);
+			});
+	};
+
+	function startup () {
+		if (isSidebar && useLocation().pathname === "/announcements") {
+			isSidebar = false;
+		}
+		const locationHash = window.location.hash.split("#");
+		if (locationHash[1] && editing === "") {
+			setEditing(locationHash[1]);
+		}
+	}
+	startup();
+
+	function AddAnnouncement() {
+		return <Box>
+			<Typography variant="h5">Add new announcement</Typography>
+			<InputFields
+				titleComponenId="titleInputText"
+				contentComponenId="contentInputText"
+			/>
+			<Table>
+				<TableBody>
+					<TableRow>
+						<TableCell style={{ color: "white", borderBottom: "none" }}>
+							<Button
+								onClick={handleClearButton}
+								fullWidth
+								variant="contained"
+								sx={{ mt: 3, mb: 2 }}
+							>
+								Clear
+							</Button>
+						</TableCell>
+						<TableCell style={{ color: "white", borderBottom: "none" }}>
+							<Button
+								onClick={handleSubmitButton}
+								fullWidth
+								variant="contained"
+								sx={{ mt: 3, mb: 2 }}
+							>
+								Submit
+							</Button>
+						</TableCell>
+					</TableRow>
+				</TableBody>
+			</Table>
+		</Box>;
+	}
+
+	return <Container component="main" maxWidth="md">
+		{userIsAdmin && !isSidebar
+			?
+			<>
+				<AddAnnouncement />
+				{errorMessageAddAnnouncement !== "" && <Grid item xs={12}>
```

Comment:
Because conditional alert comes with a state, like this:

 `const [errorMessageGetAnnouncement, setErrorMessageGetAnnouncement] = useState("");`

I already had a handful of states to manage, so I just keep the old method here.

Reply:
The good way would have been to group multiple state variables into bigger ones.